### PR TITLE
Migrate to Github Actions

### PR DESCRIPTION
**Pull request by Xorima**
# Description

This commit will migrate the build system to github actions

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

## Check List

- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        uses: actionshub/markdownlint@master
+
+  dokken:
+    needs: [mdl, yamllint, delivery]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os:
+          - 'debian-8'
+          - 'debian-9'
```
Comment (tas50) [inline .github/workflows/ci.yml]: Can we swap debian 8 for 10 while we're here

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          - 'centos-7'
```
Comment (tas50) [inline .github/workflows/ci.yml]: and add centos 8

Comment (Xorima) [reply .github/workflows/ci.yml]: wow that response time :D 

Let's see if it all works first then will make tweaks as required, got ideas for a couple other workflows as well around touched the kitchen files and not this file and touched a workflow so reviewer please ensure all changes are correct, your thoughts?

File: .github/workflows/branchcleanup.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+env:
+  NO_BRANCH_DELETED_EXIT_CODE: 0
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: jessfraz/branch-cleanup-action@master
```
Comment (damacus) [inline .github/workflows/branchcleanup.yml]: There is litteraly a tick box in the github UI fo this now.

File: libraries/helpers.rb [hunk 4]
```diff
@@ -82,8 +82,7 @@ def lib_dir
       end
 
       def libexec_dir
-        case node['platform_family']
-        when 'freebsd', 'suse'
+        if platform_family?('freebsd') || platform_family?('suse')
```
Comment (Xorima) [inline libraries/helpers.rb]: please fix my tests that all go bang ... :(

File: .github/workflows/branchcleanup.yml (hunk 3 above)
Comment (Xorima) [reply .github/workflows/branchcleanup.yml]: problem is you need admin access and they haven't exposed it to the API, which means terraform cannot do it for us.

This at least makes it more obvious to people as to how we are doing it and does not require an admin remembering something (We already have that with sponsers, I don't want to be the bottleneck to lots of things)

**Comment by damacus** [comment]
Looks like the default-dokken-debian-8 is actually running way more than debian-8 :/

**Comment by Xorima** [comment]
good catch @damacus , looks like something isn't feeding in on the action properly: `/usr/bin/kitchen test -`, code is in actionshub org so if you want to take a look feel free else I will later tonight

**Comment by Xorima** [comment]
this also includes fixes for #669

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Xorima) [reply .github/workflows/ci.yml]: Will create as a separate project for others to do, lets keep this focused on the build system else scope creep gets painful

**Pull request merged**